### ISSUE TITLE
Make it possible to to get Grants from hoard API

### ISSUE THREAD

**Issue by compleatang** [monax/hoard#23]
We need to be able to get Grants from Hoard API. We should expose some functions that act on references and provide grants, but we might also want to provide single calls to push data and get a grant, or to unpack and repack grants from type of grant to another or from one grant recipient to another...

`Get` and `Put` act on reference and data respectively:

```
rpc Get (GetRequest) returns (GetResponse);
rpc Put (PutRequest) returns (PutResponse);
```

It should probably be possible to `Put` some data and get a `Grant` of the desired type back in a single call. Similarly we may want to `Get` a `Grant` rather than the data itself. These functions probably belong behind different API calls but we could add some options to the requests. Another possibility is to provide some sort of API-level pipelining where you can call something like `Put | OpenPGPGrant` and get a grant type back if the arguments types match (or simply on supported pairs of API call).

**Comment by compleatang** [comment]
formerly AN-239

**Comment by gregdhill** [comment]
Closed by #33
